### PR TITLE
Book jpa setup

### PR DESCRIPTION
**Pull request by vsych321**

### CODE REVIEW

File: src/main/java/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public BookDto getBookById(@PathVariable Long id) {
+        return bookService.getById(id);
+    }
+
+    @PostMapping
+    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+        return bookService.save(requestDto);
+    }
+
+    @ResponseStatus(HttpStatus.NO_CONTENT)
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/controller/BookController.java:50]: Be consistent - if you added response status for one endpoint, add for others

File: src/main/java/bookstore/repository/SpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package bookstore.repository;
+
+import org.springframework.data.jpa.domain.Specification;
+
+public interface SpecificationProvider<T> {
+
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/SpecificationProvider.java]: remove empty line

File: src/main/java/bookstore/repository/book/BookSpecificationProviderManager.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import bookstore.repository.SpecificationProvider;
+import bookstore.repository.SpecificationProviderManager;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationProviderManager implements SpecificationProviderManager<Book> {
+
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/book/BookSpecificationProviderManager.java]: remove empty line

File: src/main/java/bookstore/repository/book/BookSpecificationProviderManager.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    private List<SpecificationProvider<Book>> bookSpecificationProviders;
+
+    @Override
+    public SpecificationProvider<Book> getSpecificationProvider(String key) {
+        return bookSpecificationProviders.stream()
+                .filter(p -> p.getKey().equals(key))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException(
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/repository/book/BookSpecificationProviderManager.java]: Let's create custom exception SpecificationNotFoundException for example

File: src/main/java/bookstore/entity/Book.java [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    private String description;
+    private String coverImage;
+
+    @Column(nullable = false)
+    private boolean isDeleted = false;
+}
+
```
Comment (Elena-Bruyako) [inline src/main/java/bookstore/entity/Book.java]: remove

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

File: src/main/java/bookstore/controller/BookController.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    @DeleteMapping("{id}")
+    public void delete(@PathVariable Long id) {
+        bookService.deleteById(id);
+    }
+
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    @GetMapping("/search")
+    public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```
Comment (Andriy-dz) [inline src/main/java/bookstore/controller/BookController.java:58]: as far as I know, this endpoint does not work correctly

**Review by Andriy-dz — CHANGES_REQUESTED** [review]

File: src/main/java/bookstore/controller/BookController.java (hunk 6 above)
Comment (vsych321) [reply src/main/java/bookstore/controller/BookController.java:58]: but it's headlined in task

**Review by Andriy-dz — APPROVED** [review]

**Pull request closed without merging**